File: QuanLyKhachSanApp/QuanLyKhachSanApp/Models/BangGia.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace QuanLyKhachSanApp.Models
{
    [Table("BangGia")]
    public class BangGia
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Required]
        public int BangGiaID { get; set; }

        public int? LoaiPhongID { get; set; }

        public double? GiaPhong { get; set; }

        [Column(TypeName = "Date")]
        public DateTime? ApDungTuNgay { get; set; }

        [Column(TypeName = "Date")]
        public DateTime? ApDungDenNgay { get; set; }

        [ForeignKey("LoaiPhongID")]
        public virtual LoaiPhong LoaiPhong { get; set; }

        [InverseProperty("BangGia")]
        public virtual ICollection<ThuePhong> ThuePhong { get; set; }

    }
}

File: QuanLyKhachSanCli/src/models/BangGia.ts
import { ThuePhong } from "@/models/ThuePhong";
import { LoaiPhong } from "@/models/LoaiPhong";

export interface BangGia {
    BangGiaID: number;
    LoaiPhongID: number;
    GiaPhong: number;
    ApDungTuNgay: Date;
    ApDungDenNgay: Date;
    LoaiPhong?: LoaiPhong;
    ThuePhong?: ThuePhong[];
}
